test-common/Nettest/Holdings.hs

```diff
@@ -0,0 +1,81 @@
+-- SPDX-FileCopyrightText: 2020 TBD
+--
+-- SPDX-License-Identifier: LicenseRef-Proprietary
+module Nettest.Holdings
+  ( nettestScenario
+  ) where
+
+import Lorentz.Address
+import Morley.Nettest
+import Util.Named
+
+import Indigo.Contracts.Holdings
+import qualified Indigo.Contracts.Safelist as SL
+
+nettestScenario
+  :: forall m. Monad m
+  => NettestImpl m -> m ()
+nettestScenario = uncapsNettest $ do
+  comment "Registering addresses"
+  ownerAddr :: Address <- resolveNettestAddr
+  adminAddr :: Address <- newAddress "admin"
+  someGuyAddr :: Address <- newAddress "someGuy"
+  senderAddr :: Address <- newAddress "sender"
+  receiver :: Address <- newAddress "receiver"
+  fakeSender :: Address <- newAddress "fakeSender"
+  holdingsAddr :: Address <- originateSimple "Holdings"
+    (mkStorage ownerAddr Nothing dummyMeta) holdingsContract
+  safelistAddr :: Address <- originateSimple "DummySafelist"
+    (SL.mkStorage [(senderAddr, receiver)] [senderAddr, receiver, adminAddr])
+    SL.safelistContract
+  let
+    owner, admin, someGuy, sender :: AddrOrAlias
+    owner = AddrResolved ownerAddr
+    admin = AddrResolved adminAddr
+    someGuy = AddrResolved someGuyAddr
+    sender = AddrResolved senderAddr
+
+    holdings :: AddrOrAlias
+    holdings = AddrResolved holdingsAddr
+
+  comment "Holdings nettest scenario"
+  callFrom owner holdings (ep "transferAdminRights") adminAddr
+  expectFailure (callFrom someGuy holdings (ep "acceptAdminRights") ())
+    NettestFailedWith
+  callFrom admin holdings (ep "acceptAdminRights") ()
+  expectFailure (callFrom someGuy holdings (ep "transferAdminRights") someGuyAddr)
+    NettestFailedWith
+  callFrom admin holdings (ep "mint")
```

Comment:
Would be nice to separate these calls (with `comment`s as well), for instance all the ones above this are about admin privilege etc.